tests/foreman/performance/common/perf_concurrent_test.py

```diff
@@ -0,0 +1,372 @@
+
+    def _set_bucket_size(self):
+        """Set size for each bucket"""
+        self.bucket_size = self.num_iterations / self.num_buckets
+
+    def _join_all_threads(self, thread_list):
+        """Wait for all threads to complete
+
+        @para: thread_list
+        """
+        for t in thread_list:
+            t.join()
+
+    def _write_raw_csv_file(self, raw_file_name, time_result_dict,
+                            current_num_threads, header):
+        """ Write raw timing ak/del results to csv file"""
+        self.logger.debug("Timing result is: {}".format(time_result_dict))
+        with open(raw_file_name, 'a') as f:
+            writer = csv.writer(f)
+            writer.writerow([header])
+
+            # for each thread, write its head and data
+            for i in range(current_num_threads):
+                writer.writerow([
+                    'client-{}'.format(i)
+                ])
+                writer.writerow(time_result_dict.get('thread-{}'.format(i)))
+            writer.writerow([])
+
+    def _write_raw_att_csv_file(self, raw_file_name, time_result_dict,
+                                current_num_threads, header):
+        """Write raw timing att results to csv file"""
+        self.logger.debug("Timing result is: {}".format(time_result_dict))
+        with open(raw_file_name, 'a') as f:
+            writer = csv.writer(f)
+            writer.writerow([header])
+
+            # for each thread, write its head and data
+            for i in range(current_num_threads):
+                writer.writerow(['client-{}'.format(i)])
+                for index in range(3):
+                    writer.writerow(time_result_dict.get(
+                                    'thread-{}'.format(i))[index])
+                    writer.writerow([])
+
+    def _write_stat_csv_file(self, stat_file_name, time_result_dict,
+                             current_num_threads, header, is_attach):
+        """Generate statistical result of concurrent ak/del/att"""
+        with open(stat_file_name, 'a') as f:
+            writer = csv.writer(f)
+            writer.writerow([header])
+
+            """1) write stat-per-client-bucketized result of ak/del/att"""
+            writer.writerow(["stat-per-client-bucketized"])
+            self._write_stat_per_client_bucketized(
+                stat_file_name, time_result_dict,
+                current_num_threads, is_attach
+            )
+            writer.writerow([])
+
+            """2) write stat-per-test-bucketized result of ak/del/att"""
+            writer.writerow(["stat-per-test-bucketized"])
+            self._write_stat_per_test_bucketized(stat_file_name,
+                                                 time_result_dict, is_attach)
+            writer.writerow([])
+
+            """3) write stat-per-client result of ak/del/att"""
+            writer.writerow(["stat-per-client"])
+            self._write_stat_per_client(stat_file_name, time_result_dict,
```

Comment:
Break the line after the (

```python
self._write_stat_per_client(
    stat_file_name, time_result_dict, current_num_threads, is_attach)
```

The above will look better.

This applies for other parts of the code.